test/test_scatter_gather_ops.py

```diff
@@ -85,7 +85,7 @@ def test_gather_backward_with_empty_index_tensor(self, device, dtype, sparse_gra
         self.assertEqual(grad, expected_grad, atol=0, rtol=0)
 
     def _test_scatter_base(self, fn, *, device, dtype, is_scalar, reduction,
-                           unique_indices=True, include_self=True):
+                           unique_indices=True, include_self=True, src_smaller_than_index=False):
```

Comment:
One more thing, could you please add some sample inputs to the sample_inputs functions for the opinfos of scatter/scatter_add/scatter_reduce ([e.g. here](https://github.com/pytorch/pytorch/blob/master/torch/testing/_internal/common_methods_invocations.py#L8328)) where src is smaller than index for the case where `not requires_grad` (since we aren't supporting gradients for this case)? 

This will run more comprehensive tests (e.g. `python test/test_ops.py -k scatter` includes a test that your proposed changes work on [noncontiguous tensors](https://github.com/pytorch/pytorch/blob/master/test/test_ops.py#L575))

Reply:
Roger that!

Also, I am not sure these changes actually do work on noncontiguous tensors, I haven't tested them. I _think_ they should, if the logic works as I think it does, but I'll check.

Reply:
Testing manually with some small tensors the logic does seem to work for noncontiguous tensors, and I've added sample inputs that I _think_ do the thing, if I understood them correctly.

Reply:
Okay, I had _not_ understood it correctly actually and my logic won't work for it. I'll think about how to fix it.

Reply:
@mikaylagawarecki implemented the logic for CUDA, but as @nikitaved said we might want to go with the "intersection" interpretation of the request instead of the "wraparound" one, so I'll hold off on doing it for CPU until that's decided!